opt-dist: add an option for setting path to stage0 root

ognevny opened the pull request:
in MSYS2 we have problems with stage0 for *-gnullvm hosts because prebuilt dist tarballs will be
available starting from 1.90.0-beta. also this change helps to match bootstrap.toml config

<!-- homu-ignore:start -->
<!--
If this PR is related to an unstable feature or an otherwise tracked effort,
please link to the relevant tracking issue here. If you don't know of a related
tracking issue or there are none, feel free to ignore this.

This PR will get automatically assigned to a reviewer. In case you would like
a specific user to review your work, you can assign it to them by using

    r? <reviewer name>
-->
<!-- homu-ignore:end -->
r? Kobzol

try-job: dist-x86_64-msvc
try-job: dist-x86_64-linux

ognevny commented:
@Kobzol rebased

Kobzol reviewed (commented):
Let's try.

@bors try

Kobzol commented:
@bors2 try

ognevny commented:
oh I see

ognevny committed:
opt-dist: add an option for setting path to stage0 root

in MSYS2 we have problems with stage0 for *-gnullvm hosts because prebuilt dist tarballs will be
available starting from 1.90.0-beta. also this change helps to match bootstrap.toml config

Kobzol commented:
@bors2 try

Kobzol commented:
@bors r+ rollup
